gwcs/wcs.py

```diff
@@ -634,3 +641,235 @@ def fix_inputs(self, fixed):
+        hdr['cd1_2'] = cd[0][1]
+        hdr['cd2_1'] = cd[1][0]
+        hdr['cd2_2'] = cd[1][1]
+        hdr['sipmxerr'] = max_resid
+        hdr['siprms'] = rms
+        # # for test purposes, make sip equivalent in GWCS
+        # cdmat = np.array([[cd[0][0], cd[0][1]], [cd[1][0], cd[1][1]]])
+        # # cdmat = [[cd[0][0], cd[1][0]], [cd[0][1], cd[1][1]]]
+        # sip_poly_test_x = sip_poly_x.copy()
+        # sip_poly_test_y = sip_poly_y.copy()
+        # # sip_poly_test_x.c1_0 = 1
+        # # sip_poly_test_y.c0_1 = 1
+        # shift = (Shift(-crpix1) & Shift(-crpix2))
+        # aff = AffineTransformation2D(cdmat, [0,0])
+
+        # transform = (shift | Mapping([0,1,0,1]) |
+        #             (Mapping([0], n_inputs=2) + sip_poly_test_x) &
+        #             (Mapping([1], n_inputs=2) + sip_poly_test_y) |
+        #             aff |
+        #             Pix2Sky_TAN() | RotateNative2Celestial(crval1, crval2, 180))
+        # transform2 = ((Shift(-crpix1) & Shift(-crpix2)) | Mapping([0,1,0,1]) |
+        #               (fit_poly_x & fit_poly_y) | Pix2Sky_TAN() |
+        #               RotateNative2Celestial(crval1, crval2, 180))
+        return hdr
+
+
+def fit_2D_poly(degree, max_error, max_rms, u, v, x, y, verbose=False):
+    """
+    Fit a pair of ordinary 2D polynomial to the supplied inputs (u, v) and
+    outputs (x, y)
+    """
+    llsqfitter = LinearLSQFitter()
+    # The case of one pass with the specified polynomial degree
+    if degree is not None:
+        poly_x = Polynomial2D(degree=degree)
+        poly_y = Polynomial2D(degree=degree)
+        fit_poly_x = llsqfitter(poly_x, u, v, x)
+        fit_poly_y = llsqfitter(poly_y, u, v, y)
+        max_resid, rms = compute_distance_residual(x, y,
+                                                   fit_poly_x(u, v), fit_poly_y(u, v))
+    else:
+        cond = True
+        degree = 2
+        while cond:
+            poly_x = Polynomial2D(degree=degree)
+            poly_y = Polynomial2D(degree=degree)
+            fit_poly_x = llsqfitter(poly_x, u, v, x)
+            fit_poly_y = llsqfitter(poly_y, u, v, y)
+            max_resid, rms = compute_distance_residual(x, y,
+                                                       fit_poly_x(u, v), fit_poly_y(u, v))
+            if verbose:
+                print(f'Degree = {degree}, max_resid = {max_resid}, rms = {rms}')
+            if ((max_error is not None and max_resid < max_error and max_rms is None)
+                 or (max_error is not None and max_resid < max_error and max_rms is not None
+                     and rms < max_rms)
+                 or (max_error is None and max_rms is not None and rms > max_rms)):
+                cond = False
+                if verbose:
+                    print('terminating condition met')
+                break
+            else:
+                degree += 1
+                if degree == 10:
+                    raise RuntimeError("Fit conditions not met")
```

Comment:
I would replace this with a more informative message:
```suggestion
                    raise RuntimeError("Unable to achieve required fit accuracy using SIP polynomials of degree < 10")
```

Reply:
Made a similar change.